### PR TITLE
Automatically updating chain.json that chains can maintain

### PR DESCRIPTION
**Pull request by xBalbinus**
Closes: #425 

This PR adds Github actions and a Python script that automatically make a PR whenever chain.json on the chains' repo changes.

The bot should:
- Looks for the schema url in each chain supported
- Tries downloading latest from each chains URL
- Checks if for every chain there is a difference
- If theres a difference, make a single commit doing all the updates to a new branch
- open a PR for that branch to main.

### CODE REVIEW

File: .github/workflows/utility/update_chaindata.py
Comment (ValarDragon) [inline .github/workflows/utility/update_chaindata.py:21]: We need to update the Osmosis chain.json to get this working / test it right?

**Comment by ValarDragon** [comment]
Do you have a sample run of the workflow on your fork?

**Comment by xBalbinus** [comment]
This still needs a test run, will keep you posted!

File: .github/workflows/utility/update_chaindata.py
Comment (xBalbinus) [reply .github/workflows/utility/update_chaindata.py:21]: I don't think so, since the current implementation of the chain.json on our end is just for the "codespace" part of the chain.json schema on the chian-registry. I'll share a working run soon.

Comment (xBalbinus) [reply .github/workflows/utility/update_chaindata.py]: Sorry unclear answer on my part - the osmosis chain.json has already been updated, no further action req here.

File: .github/workflows/utility/update_chaindata.py
Comment (ValarDragon) [inline .github/workflows/utility/update_chaindata.py:29]:
```suggestion
                    json.dump(current, f, ensure_ascii=False, indent=4)
```

**Comment by ValarDragon** [comment]
Can you try the fix I posted, and then paste a CI run here, and the diff update applied?

File: .github/workflows/update_chainjson.yml
Comment (ValarDragon) [inline .github/workflows/update_chainjson.yml:3]: Should be uncommented right?

**Comment by xBalbinus** [comment]
Will do! And yes the below will be uncommented but has been commented at
the moment for testing

File: .github/workflows/utility/update_chaindata.py
Comment (xBalbinus) [reply .github/workflows/utility/update_chaindata.py]: Thanks so much! lol I totally misunderstood the fix when we were talking earlier, really appreciate this

**Comment by xBalbinus** [comment]
PR Successfully created: Screen Shot 2022-06-21 at 10 33 10 AM

**Comment by xBalbinus** [comment]
The successful run: https://github.com/xBalbinus/chain-registry/actions/runs/2536132975
Created PR: https://github.com/xBalbinus/chain-registry/pull/1

**Comment by xBalbinus** [comment]
Confirmed, second run after initial change was merged does not create a PR.
https://github.com/xBalbinus/chain-registry/actions/runs/2545337422

**Review by ValarDragon — APPROVED** [review]
Lets restore the schedule, merge this in, and see if does the compatible versions update tomorrow!

File: .github/workflows/update_chainjson.yml
Comment (ValarDragon) [inline .github/workflows/update_chainjson.yml]: Lets keep the manual workflow dispatch as well!

Comment (ValarDragon) [reply .github/workflows/update_chainjson.yml]: (make it run on both a schedule OR manual)

**Comment by ValarDragon** [comment]
Can you fix CI?

**Comment by xBalbinus** [comment]
CI Fixed and retested, both successful.

1st run: https://github.com/xBalbinus/chain-registry/actions/runs/2549405299
2nd run: https://github.com/xBalbinus/chain-registry/actions/runs/2549425795

**Pull request merged**
